ebib.el

```diff
@@ -3796,14 +3802,16 @@ was called interactively."
              (field (completing-read "Jump to field: " fields nil t))
              (location (save-excursion
                          (goto-char (point-min))
-                         (re-search-forward (concat "^" field)))))
-    (goto-char location)
-    (beginning-of-line)))
+                         (re-search-forward (concat "^" field) nil t))))
+    (if location
+	(goto-char location)
+      (beginning-of-line))))
 
-(defun ebib-add-field (field)
+(defun ebib-add-field ()
   "Add FIELD to the current entry."
```

Comment:
So as above, ivy actions actually need to take exactly one argument. That alone makes this a bug, because the function *doesn't* take any arguments.

The *reason* it doesn't take any arguments is that I had thought calling a function from an ivy action was a non-interactive use so I needed to make sure that the function prompted for a string whatever the commit message was (I'm really glad I wrote a good commit message for that one!).

In fact, using the `(interactive "s...")` as before *does* work (presumably ivy accounts for this somehow). What happens is that the current input is used as the `field` arg. This makes sense workflow-wise: if you try to jump to a nonexistent field, you probably want to add that field. However, there once you hit <kbd>a</kbd> to add a new field, there is no opportunity to amend or add the input for the field name, you are just prompted for a value. I don't think this is quite what we want. For me it's most natural to use ivy's regex matching to see that a certain field isn't present, *then* do the actoin to add. I would expect to be prompted, perhaps with my previous input as a defualt in the prompt? What do you think? (Not that it has anything to do with ivy, but I wonder if it might be useful to have add-field work a bit like the keyword prompt, allowing new additions, but also selecting from a list of everything *already* added in the database? Just a thought.)